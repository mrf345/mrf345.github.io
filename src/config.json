{
    "name": "Mohamed Feddad",
    "occupation": "Software Engineer",
    "account": "mrf345",
    "repo": "https://github.com/mrf345/mrf345.github.io/tree/testing",
    "profileImage": "https://avatars1.githubusercontent.com/u/26286907?s=460&v=4",
    "description": [
        "I'm an experienced freelance software engineer mostly in fullstack web development working in various industries with a variety of web stacks and architectures.",
        "I'm very passionate about software production as whole, and working with Python, TypeScript and Golang. I enjoy developing clean, tested, and well-documented software,",
        " and contributing to open-source projects as a hobby."
    ],
    "intro": [
        "I'm Mo (Mohamed Feddad) a fullstack engineer, passionate about programming and software production as a whole.",
        "I love open-source development and what it stands for, empowering others and improving one's skills openly. I mostly program in",
        "Python, TypeScript, and Go."
    ],
    "experienceInKeywords": [
        "Python3/2", "Django", "Django-Rest-Framework", "Web.py", "Flask", "Flask-Restx", "Flask-Migration", "Flask-SQLAlchemy", "Swagger", "Jinja2", "Gevent", "Asyncio",
        "PyCrypto", "SQLAlchemy", "MySQL", "MariaDB", "PostgreSQL","SQLite3", "SQLCipher", "MongoDB", "PyQt4/5", "PySide", "PyInstaller", "BeautifulSoup", "Unittest", "PyTest",
        "Selenium",  "JavaScript ES5-9", "TypeScript", "jQuery", "jQueryUI", "Knockout.js", "LESS", "SASS", "Bootstrap", "Node.js", "Express", "Socket.io", "Webpack", "Babel",
        "Vue.js", "Vuex", "Vue-Router", "ElementUI", "React", "Redux", "Redux-Saga", "React-Bootstrap", "React-Router", "Puppeteer", "NW.js", "Cheerio", "Jest", "Mocha", "GNU/Linux",
        "Bash Script", "Batch Script", "Docker", "Docker-Compose", "AWS EC2", "AWS Lambda", "AWS RDS", "FireBase", "FireStore", "Google Cloud Functions", "Github Actions", "Jenkins",
        "Travis-CI", "Sentry", "Jira", "Github Projects", "Agile", "Scaled Agile"
    ],
    "references":[
        ["EMail", "mailto://moe.feddad@gmail.com", "moe.feddad@gmail.com"],
        ["Github", "https://github.com/mrf345", "github.com/mrf345"],
        ["Address", "", "Dubai, United Arab Emirates."]

    ]
}
