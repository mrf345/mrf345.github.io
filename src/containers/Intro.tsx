import React from 'react'
import { connect } from 'react-redux'

import Panel from '../components/Panel'
import info from '../config.json'


interface InfoProps {
    info:Info
    loading:Loading
}


class Intro extends React.Component<InfoProps> {
    componentDidMount() {
        window.scrollTo(0, document.body.scrollHeight)
    }

    render() {
        return <Panel className='text-center d-flex align-items-center min-vh-100 vw-100'
                   goto={`${window.location.origin}/cv.pdf`}
                   clickAnimation={{animation: 'bounceOut', delay: 0}}
                   img={{src: this.props.info.avatar_url,
                         width: '30%'}}
                   animation={{animationIn: 'bounceIn',
                               animationInDelay: 500,
                               animationOut: 'bounceOut',
                               isVisible: true}}
                   header={{text: <>{this.props.info.name} <br /> <i>({this.props.info.login})</i></>,
                            className: 'h2 mt-2'}}
                   footer={{center: 'Click to view my resume!'}}
                   body={{text: info.description.join(' '),
                          style: {fontSize: '120%'},
                          className: 'mt-4 mb-4'}} />
    }
}


export default connect((state:State) => ({info: state.info,
                                          loading: state.loading}))(Intro)
