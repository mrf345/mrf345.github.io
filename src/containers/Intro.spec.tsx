import React from "react"
import { Provider } from "react-redux"
import { render } from "@testing-library/react"
import configureStore from "redux-mock-store"

import Intro from "./Intro"
import config from "../config.json"


describe("Testing Intro container", () => {
  const mockStore = configureStore([])
  const name = "Testing Name"
  const avatar_url = "http://somewhere.testing"
  const login = "Testing Login"
  let store: any

  beforeEach(() => {
    store = mockStore({info: {name, avatar_url, login},
                       loading: {info: false}})
  })

  test("Sanity check Intro content", () => {
    const component = render(
      <Provider store={store}>
        <Intro />
      </Provider>
    )
    const description = component.getByText(config.description.join(" "))

    expect(description).toBeTruthy()
    expect(description.tagName).toEqual("DIV")

    description.remove()
    expect(component).toMatchSnapshot()
  })
})
