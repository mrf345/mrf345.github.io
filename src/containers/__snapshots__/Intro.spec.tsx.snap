// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Intro container Sanity check Intro content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-center d-flex align-items-center min-vh-100 vw-100 col-lg col-md-10 offset-md-1"
        footer="[object Object]"
      >
        <div
          class="animated bounceIn"
          style="animation-delay: 500ms; animation-duration: 1000ms; pointer-events: all;"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <div
                class="col"
              />
            </div>
            <div
              class="card-body"
            >
              <div
                class="mb-2 col"
              />
              <div
                class="mt-4 mb-4 col"
                style="font-size: 120%;"
              >
                <h1
                  class="cursored"
                >
                  Hello, world...
                </h1>
                <div
                  class="cursor"
                />
                <div
                  class="mt-5 intro-text"
                >
                  You can navigate to my open-source 
                  <a
                    class="intro-nav-link"
                    href="#/projects"
                  >
                    Projects
                  </a>
                  , my 
                  <a
                    class="intro-nav-link"
                    href="#/contributions"
                  >
                    Contributions
                  </a>
                   to other projects,
                  <br />
                </div>
                <div
                  class="mt-2 intro-text"
                >
                  and if you're not a stalker feel free to browse my 
                  <a
                    class="intro-nav-link"
                    href="http://localhost/cv.pdf"
                    target="_link"
                  >
                    Resume
                  </a>
                </div>
              </div>
            </div>
            <div
              class="card-footer"
            >
              <div
                class="col"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-center d-flex align-items-center min-vh-100 vw-100 col-lg col-md-10 offset-md-1"
      footer="[object Object]"
    >
      <div
        class="animated bounceIn"
        style="animation-delay: 500ms; animation-duration: 1000ms; pointer-events: all;"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <div
              class="col"
            />
          </div>
          <div
            class="card-body"
          >
            <div
              class="mb-2 col"
            />
            <div
              class="mt-4 mb-4 col"
              style="font-size: 120%;"
            >
              <h1
                class="cursored"
              >
                Hello, world...
              </h1>
              <div
                class="cursor"
              />
              <div
                class="mt-5 intro-text"
              >
                You can navigate to my open-source 
                <a
                  class="intro-nav-link"
                  href="#/projects"
                >
                  Projects
                </a>
                , my 
                <a
                  class="intro-nav-link"
                  href="#/contributions"
                >
                  Contributions
                </a>
                 to other projects,
                <br />
              </div>
              <div
                class="mt-2 intro-text"
              >
                and if you're not a stalker feel free to browse my 
                <a
                  class="intro-nav-link"
                  href="http://localhost/cv.pdf"
                  target="_link"
                >
                  Resume
                </a>
              </div>
            </div>
          </div>
          <div
            class="card-footer"
          >
            <div
              class="col"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
